shell/shared/shell.js

```diff
@@ -935,6 +1008,16 @@ Router.map(function () {
     }
   });
 
+  this.route("referrals", {
+    path: "/referrals",
+
+    waitOn: function() {
+      return [Meteor.subscribe("myReferrals"),
```

Comment:
Note that a publish can actually return multiple cursors. So in theory you could combine these three subscribes into one big subscribe. However, last I knew Meteor had a known bug that it couldn't support a single publish returning multiple cursors from the same collection -- and all your cursors are from the users collection. Does that restriction still exist? I believe if you try it it will give a clear error if so.

Hmm, another thing to consider here is whether you really ought to be subscribing to a pseudo-collection of some sort. It's mildly disconcerting to me that we're returning cursors into the users table, matching users other than the caller. You've been careful to select only safe fields, but it still makes me worry since this is the most sensitive table in the database.

Taking into consideration all the comments above, perhaps it would actually make a lot of sense to have a single publish which publishes a pseudo-collection which is populated by the identities Alice has referred and a boolean "completed" flag for each one. On the server side, one query is performed to find all identities with `referredBy` equals Alice (and we stop removing `referredBy` from identities on completion, so this matches completed referrals too), and then individual queries are done (still on the server side) to each account to see if it is completed, in order to fill in the boolean fields.